### DISCUSSION TITLE
add option to export beatmaps in osu file format v14

### DISCUSSION THREAD

**Discussion by nanashi-1** [ppy/osu#19496] [Ideas] [answered]
Exporting maps created using the osu!lazer editor results in an osu file with format v128 which in my experience doesn't work with osu!stable. This makes uploading maps created using the osu!lazer editor extremely difficult.

Below is the osu file for the map I created using the osu!lazer editor.

```ini
﻿osu file format v128

[General]
AudioFilename: yuri.mp3
AudioLeadIn: 0
PreviewTime: 51329
Countdown: 1
SampleSet: Normal
StackLeniency: 0.5
Mode: 0
LetterboxInBreaks: 0
WidescreenStoryboard: 1

[Editor]
DistanceSpacing: 1
BeatDivisor: 4
GridSize: 32
TimelineZoom: 1.5

[Metadata]
Title: Yuriyurarararayuruyuri Daijiken
TitleUnicode: ゆりゆららららゆるゆり大事件
Artist: Nanamori-chu*Gorakubu
ArtistUnicode: 七森中☆ごらく部
Creator: nanasi20040817
Version: Insane
Source: ゆるゆり
Tags: yuru yuri

[Difficulty]
HPDrainRate: 4.5
CircleSize: 4
OverallDifficulty: 7
ApproachRate: 9
SliderMultiplier: 1
SliderTickRate: 1

[Events]
0,0,"BG.jpg",0,0

[TimingPoints]
some timing data

[Colours]
Combo0: 119,143,227,255
Combo1: 222,61,61,255
Combo2: 75,233,87,255
Combo3: 251,172,85,255

[HitObjects]
some hit object data
```

And below is the osu file after being converted to v14

```ini
osu file format v14

[General]
AudioFilename: audio.mp3
AudioLeadIn: 0
PreviewTime: 51329
Countdown: 0
SampleSet: None
StackLeniency: 0.7
Mode: 0
LetterboxInBreaks: 0
WidescreenStoryboard: 0

[Editor]
DistanceSpacing: 3.1
BeatDivisor: 4
GridSize: 32
TimelineZoom: 1

[Metadata]
Title:Yuriyurarararayuruyuri Daijiken
TitleUnicode:ゆりゆららららゆるゆり大事件
Artist:Nanamori-chu*Gorakubu
ArtistUnicode:七森中☆ごらく部
Creator:nanasi20040817
Version:Insane
Source:ゆるゆり
Tags:yuru yuri
BeatmapID:3729410
BeatmapSetID:1817773

[Difficulty]
HPDrainRate:4.5
CircleSize:4
OverallDifficulty:7
ApproachRate:9
SliderMultiplier:1
SliderTickRate:1

[Events]
//Background and Video events
0,0,"BG.jpg",0,0
//Break Periods
//Storyboard Layer 0 (Background)
//Storyboard Layer 1 (Fail)
//Storyboard Layer 2 (Pass)
//Storyboard Layer 3 (Foreground)
//Storyboard Layer 4 (Overlay)
//Storyboard Sound Samples

[TimingPoints]
some timing data


[Colours]
Combo1 : 222,61,61
Combo2 : 75,233,87
Combo3 : 251,172,85
Combo0 : 119,143,227

[HitObjects]
some hit object data
```

Converting may sound like a solution, and yes if you intend to run it only in osu!stable because it is broken in osu!lazer.

Below are screenshots of the problem of the map I created after I converted it to v14, uploaded it, then downloaded it in osu!lazer.

![Verify Tab after conversion](https://user-images.githubusercontent.com/54083498/182022428-9ada7f9d-5737-4363-b51f-81aaf19a0d83.png)

![Slider speed is now 1.00x](https://user-images.githubusercontent.com/54083498/182022438-c79e14cd-713f-4d00-8c45-348406c8ce53.png)

![Unsnapped slider](https://user-images.githubusercontent.com/54083498/182022453-64445153-6c30-47ad-a9da-358c7f6f97d7.png)

As you can see, the timing of the sliders is now messed up.

I suggest adding an option to export a beatmap in v14 format to maybe fix this issue or add a way to upload maps directly in osu!lazer.

**Comment by peppy** [comment, marked as answer]
There's an eventual plan to support this process, by either conversion or an update to stable.
